lib/dfe_sign_in/user_access.rb

```diff
@@ -0,0 +1,47 @@
+module DfeSignIn
+  class UserAccess
+    include DfeSignIn::Utils
+
+    attr_accessor :organisation_id,
+                  :user_id
+
+    def initialize(organisation_id:, user_id:)
+      self.organisation_id = organisation_id
+      self.user_id = user_id
+    end
+
+    def has_role?(role)
+      roles.include?(role)
+    end
+
+    private
+
+    def body
+      @body ||= begin
+        request = Net::HTTP::Get.new(uri)
+        request["Authorization"] = "bearer #{generate_jwt_token}"
+        request["Content-Type"] = "application/json"
+
+        response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) { |http|
+          http.request(request)
+        }
+
+        raise ExternalServerError if response.code.eql?("500")
+
+        JSON.parse(response.body) if response.code.eql?("200")
```

Comment:
If the response code isn't "500" or "200" this method will return/assign `nil`, which will be make it hard to debug and track down issues as we will have no trace of the what the response was from the API. Instead, perhaps we could raise in _all_ cases, except when the response is 200. Also, instead of just `raise ExternalServerError`, can we include the code and body as a message for the exception to aid debugging? For example, something like `raise ExternalServerError, "#{response.code}: #{response.body}"`